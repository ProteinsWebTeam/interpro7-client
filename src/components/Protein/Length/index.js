// @flow
import React, {PropTypes as T} from 'react';
import styles from 'styles/blocks.css';


const Length = ({metadata, pathname}) => {
  return (
    <div>Length: {metadata.length} amino acids</div>
  )};

Length.propTypes = {
  metadata: T.shape({
    length: T.number.isRequired
  }).isRequired,
  pathname: T.string.isRequired,
};

export default Length;
