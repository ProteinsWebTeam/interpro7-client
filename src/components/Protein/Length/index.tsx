import React from 'react';

import Link from 'components/generic/Link';

type Props = {
  metadata: {
    length: number;
    fragment?: string;
  };
};

const Length = ({ metadata: { length, fragment } }: Props) => {
  let fragmentText;
  if (fragment) {
    fragmentText = ` (${fragment === 'N' ? 'complete' : 'fragment'})`;
  }
  return (
    <Link
      to={({ description }) => ({
        description: {
          ...description,
          [description.main.key]: {
            ...description[description.main.key],
            detail: 'sequence',
          },
        },
      })}
    >
      {length} amino acids
      {fragmentText}
    </Link>
  );
};

export default Length;
