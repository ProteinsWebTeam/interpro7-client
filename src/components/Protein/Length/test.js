// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Length from '.';

const renderer = new ShallowRenderer();

describe('<Length />', () => {
  test('should render', () => {
    renderer.render(<Length metadata={{ length: 100, fragment: 'N' }} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    renderer.render(<Length metadata={{ length: 100, fragment: 'Y' }} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
    renderer.render(<Length metadata={{ length: 100 }} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
