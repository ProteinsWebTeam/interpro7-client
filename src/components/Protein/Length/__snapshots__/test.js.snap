// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Length /> should render 1`] = `
<Memo(ConnectFunction)
  to={[Function]}
>
  100
   amino acids
   (complete)
</Memo(ConnectFunction)>
`;

exports[`<Length /> should render 2`] = `
<Memo(ConnectFunction)
  to={[Function]}
>
  100
   amino acids
   (fragment)
</Memo(ConnectFunction)>
`;

exports[`<Length /> should render 3`] = `
<Memo(ConnectFunction)
  to={[Function]}
>
  100
   amino acids
</Memo(ConnectFunction)>
`;
